packages/base-manager/src/manager-base.ts

```diff
@@ -476,6 +477,42 @@ export abstract class ManagerBase implements IWidgetManager {
     moduleVersion: string
   ): Promise<typeof WidgetModel | typeof WidgetView>;
 
+  protected async loadModelClass(
+    className: string,
+    moduleName: string,
+    moduleVersion: string
+  ): Promise<typeof WidgetModel> {
+    try {
+      const promise: Promise<typeof WidgetModel> = this.loadClass(
+        className,
+        moduleName,
+        moduleVersion
+      ) as Promise<typeof WidgetModel>;
+      await promise;
+      return promise;
```

Comment:
Why do you need to await the promise before returning it?

Reply:
to catch a possible exception